### PR TITLE
8253433: Remove -XX:+Debugging product option

### PR DESCRIPTION
**Pull request by coleenp**
The Debugging option shouldn't be used on the command line.  There's a SuppressErrorAt option to ignore certain asserts, if there is some situation needing that. Debugging should never be used.

Tested with tier1 tests on 4 platforms.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253433](https://bugs.openjdk.java.net/browse/JDK-8253433): Remove -XX:+Debugging product option

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/434/head:pull/434`
`$ git checkout pull/434`

### CODE REVIEW

**Comment by coleenp** [comment]
CSR: https://bugs.openjdk.java.net/browse/JDK-8253501
bug (I don't see linked here): https://bugs.openjdk.java.net/browse/JDK-8253433

**Comment by coleenp** [comment]
Used the wrong bug number.

File: src/hotspot/share/utilities/debug.cpp [hunk 1]
```diff
@@ -71,6 +71,9 @@ static intx g_asserting_thread = 0;
 static void* g_assertion_context = NULL;
 #endif // CAN_SHOW_REGISTERS_ON_ASSERT
 
+// Set to suppress secondary error reporting.
+bool Debugging = false;
```
Comment (tstuefe) [inline src/hotspot/share/utilities/debug.cpp:75]: How would this work? You would set this while debugging? Does it also work in optimized coding or would the value getting optimized away? Maybe make volatile?

**Review by kimbarrett — APPROVED** [review]

**Review by tstuefe — APPROVED** [review]
LGTM

**Review by dholmes-ora — APPROVED** [review]
Seems fine. Thanks.

**Comment by coleenp** [comment]
>This is really just about the one existing assignment.
See the Command context class in debug.cpp. This variable probably shouldn?t be set/reset
much of anywhere else.

There's also a place in macroAssembler that sets it to true also, that calls os::print_location() where we also want to prevent crashes.  We could rewrite that to use the debug.cpp find functions but doesn't seem worth doing.

Thank you for the code reviews.

**Pull request closed without merging**